### PR TITLE
Split Agent code into the Client package.

### PR DESCRIPTION
**Pull request by paultag**

### CODE REVIEW

**Comment by paultag** [comment]
This MR uses the encoded Host field, and finds all valid Certificates, allowing the `ssh-add` command to be invoked very frequently, without flooding the running agent with new keys.

File: client/agent.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			continue
+		}
+
+		host, ok := cert.Extensions["hallow-host@dc.cant.vote"]
+		if !ok {
+			// not a hallowed Certificate
+			continue
+		}
+
+		if strings.Compare(host, uri.Host) != 0 {
```
Comment (alex) [inline client/agent.go]: This is just `host == uri.Host`, isn't it?

Comment (paultag) [reply client/agent.go]: That's true! Thanks! I always forget when that works and when it doesn't. Fixed!

Comment (paultag) [reply client/agent.go]: (except !=, but yes)

**Comment by paultag** [comment]
@alex 

Now, given `~/.ssh/config` of 

```
Match host hackhack exec "hallow-cli --log-level=info ssh-add"
```

the following happens:

```
[paultag@nyx:~][08:29 PM] ♥  ssh-add -L | grep cert | wc -l
0
[paultag@nyx:~][08:29 PM] ♥  ssh hackhack whoami
ubuntu
[paultag@nyx:~][08:29 PM] ♥  ssh-add -L | grep cert | wc -l
1
```

**Comment by alex** [comment]
Nice. That's awfully slick.

**Pull request merged**